### PR TITLE
added ability to save `nclx` profiles

### LINKED ISSUE #118: How to save "nclx_profile" in a HEIC image?
I'm trying to add the "nclx_profile" metadata to a new HEIC file.
I'm setting this key as a dict in `pil_image.info`, but when I save and load the photo the data isn't there.

Full example code to reproduce:

```python
import numpy as np
import pillow_heif
from PIL import Image
from io import BytesIO

from pillow_heif import register_heif_opener
register_heif_opener()

nclx_profile = {
 'color_primaries': 1,
 'transfer_characteristics': 2,
 'matrix_coefficients': 1,
 'full_range_flag': 1,
 'color_primary_red_x': 0.0,
 'color_primary_red_y': 0.0,
 'color_primary_green_x': 0.0,
 'color_primary_green_y': 0.0,
 'color_primary_blue_x': 0.0,
 'color_primary_blue_y': 0.0,
 'color_primary_white_x': 0.0,
 'color_primary_white_y': 0.0
}

# create a new, blank image
pil_image = Image.fromarray(np.ones((16, 16, 3)).astype(np.uint8))
# sset nclx_profile
pil_image.info['nclx_profile'] = nclx_profile
# save to an in-memory file (works if we save to disk too)
tmp = BytesIO()
pil_image.save(tmp, format="heif")
# re-open the image and see if nclx_profile is there
Image.open(tmp).info['nclx_profile']    # KeyError!
```

### PR DESCRIPTION
**Pull request by bigcat88**
Resolves this: #118 by adding the ability to save `nclx` profiles.

### CODE REVIEW

File: pillow_heif/options.py [hunk 1]
```diff
@@ -48,3 +48,12 @@
 To learn more read: `here <https://github.com/strukturag/libheif/issues/784>`_
 
 When use pillow_heif as a plugin you can set it with: `register_*_opener(allow_incorrect_headers=True)`"""
+
+
+SAVE_NCLX_PROFILE = False
+"""Should be `nclx` profile saved or not.
+
+.. note:: macOS viewer ``does not support opening`` images with **nclx** profiles, so this is disabled by default.
```
Comment (wiggin15) [inline pillow_heif/options.py:56]: Hi. Did you test this? The HEIC file I was commenting about in #118 does show correctly in the macOS "Preview" app and in Finder "quick look" feature, even though it has nclx information set on two image streams. Where can I find the documentation of the `macOS_compatibility_workaround_no_nclx_profile` flag?

File: pillow_heif/options.py [hunk 2]
```diff
@@ -48,3 +48,12 @@
...
+.. note:: `save_nclx_profile` specified during calling ``save`` has higher priority then this.
```
Comment (wiggin15) [inline pillow_heif/options.py:57]: typo: then -> than

Comment (bigcat88) [reply pillow_heif/options.py]: thanks, fixed

File: pillow_heif/options.py (hunk 1 above)
Comment (bigcat88) [reply pillow_heif/options.py:56]: https://github.com/strukturag/libheif/blob/bcb2497474630263cf481539b25088679ae647e2/libheif/context.cc#L2520-L2523

https://github.com/strukturag/libheif/blob/bcb2497474630263cf481539b25088679ae647e2/libheif/heif.cc#L2841-L2856


Unfortunately, documentation is not the strong side of `libheif`.
And yeah, I tested today on last macOS(13.4) - Viewer was able to open a test photo.
Need to think what can be putted into description, probably better to ask in `libheif` repo

Comment (bigcat88) [reply pillow_heif/options.py:56]: I asked about it here: https://github.com/strukturag/libheif/discussions/931

Comment (bigcat88) [reply pillow_heif/options.py:56]: @wiggin15 https://github.com/strukturag/libheif/discussions/931#discussioncomment-6520340

**Comment by bigcat88** [comment]
About the topic: to save or not by default "nclx" profile

This version that will be released soon will not do this by default, but the next or one version later will most likely already be.

Reason: my current version of libheif 1.16.2 with this option does not pass all tests (it crashes inside libheif color conversion for 12-bit images when encoding a lot of them in cycle), on the master version from the repository - everything works fine.

Therefore, as soon as the next version of libheif is released with support for opening images in Grayscale format, I will rewrite half of the tests for that, behavioral compatibility will be partially broken(but not for Pillow plugin mode), and we can immediately change `SAVE_NCLX_PROFILE` default value to True.

**Pull request merged**